### ISSUE TITLE
Microsoft dhcp-ipv4 decoder: regex for srcip extracts to much data from log

### ISSUE THREAD

**Issue by VJakoby** [wazuh/wazuh#25285]
|Wazuh version| Component | Action type |
|---| --- | --- |
| 4.7.4 | Decoders | Improve |

## Description
The default decoder for microsoft dhcp "ms-dhcp-ipv4" extracts too much data on the srcip field.
It does not stop at the end of the IP address, and extracts the rest of the log.
When evaluating the regex on a regex-tester, the group 3 can be seen with more information than only the IP.

### Service/Product/Module

Decoder: ms-dhcp-ipv4
https://github.com/wazuh/wazuh-ruleset/blob/master/decoders/0380-windows_decoders.xml
Regex: ^(\d\d),\d+/\d+/\d\d\d*,\d+:\d+:\d+,(\w+),(\S+)

### Current results
data.srcip: 192.168.10.205,6ftya91701.domain.local,000B97A0B41D, (example)
![2024-08-20_14-27](https://github.com/user-attachments/assets/f454a240-d2c1-4358-8320-c2022d9d6d3d)

### Expected results

data.srcip: 192.168.10.205

### Log source / integration
Microsoft DHCP

### Log examples

15,3/10/2009,8:49:10,NACK,192.168.10.205,,000B97A0B41D,
10,3/10/2009,8:49:10,Assign,192.168.10.205,6ftya91701.domain.local,000B97A0B41D,
12,3/10/2009,15:52:38,Release,192.168.10.205,6ftya91701.domain.local.,000B97A0B41D,

**Comment by VJakoby** [comment]
![2024-08-20_14-44](https://github.com/user-attachments/assets/c6d5489c-1c7f-4b08-bfe5-a894c4bd6e56)
This is how it looks in Wazuh. (Masked IP and MAC due to privacy reasons)

**Comment by sakib789** [comment]
As a workaround for now, you can also modify the default decoder 

https://documentation.wazuh.com/current/user-manual/ruleset/decoders/custom.html#modify-default-decoders

```
<decoder name="ms-dhcp-ipv4">
  <prematch>^\d\d,\d+/\d+/\d\d\d\d,\d+:\d+:\d+,|</prematch>
  <prematch>^\d\d,\d+/\d+/\d\d,\d+:\d+:\d+,</prematch>
  <regex type="pcre2">^(\d\d),\d+/\d+/\d\d\d*,\d+:\d+:\d+,(\w+),(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\.*</regex>
  <order>id,extra_data,srcip</order>
</decoder>
```

**Comment by VJakoby** [comment]
> As a workaround for now, you can also modify the default decoder
> 
> https://documentation.wazuh.com/current/user-manual/ruleset/decoders/custom.html#modify-default-decoders
> 
> ```
> <decoder name="ms-dhcp-ipv4">
>   <prematch>^\d\d,\d+/\d+/\d\d\d\d,\d+:\d+:\d+,|</prematch>
>   <prematch>^\d\d,\d+/\d+/\d\d,\d+:\d+:\d+,</prematch>
>   <regex type="pcre2">^(\d\d),\d+/\d+/\d\d\d*,\d+:\d+:\d+,(\w+),(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\.*</regex>
>   <order>id,extra_data,srcip</order>
> </decoder>
> ```

Absolutely. Will work great. The problem will reoccur if we will do an upgrade and the issue is not fixed by default. So i wanted to let you know so its fixed in your end instead of manually needing to change it after each update.

**Comment by molenear** [comment]
On 4.14.5-1, I was able to resolve this:

1. edit 0380-windows_decoders.xml
2. find the ms-dhcp-ipv4 section (line 138)
3. find the regex section, it should look like `  <regex>^(\d\d),\d+/\d+/\d\d\d*,\d+:\d+:\d+,(\w+),(\S+)</regex>`
4. Add a comma to the end so it looks like `  <regex>^(\d\d),\d+/\d+/\d\d\d*,\d+:\d+:\d+,(\w+),(\S+),</regex>`

The final section should look like this:
`<decoder name="ms-dhcp-ipv4">`
`  <prematch>^\d\d,\d+/\d+/\d\d\d\d,\d+:\d+:\d+,|</prematch>`
`  <prematch>^\d\d,\d+/\d+/\d\d,\d+:\d+:\d+,</prematch>`
`  <regex>^(\d\d),\d+/\d+/\d\d\d*,\d+:\d+:\d+,(\w+),(\S+),</regex>`
`  <order>id,extra_data,srcip</order>`
`</decoder>`

Of course that will be overwritten with the next upgrade, but it's a simple enough fix.
